app/src/main/java/ch/epfl/sdp/blindly/chat/ChatActivity.kt

```diff
@@ -0,0 +1,143 @@
+package ch.epfl.sdp.blindly.chat
+
+import android.os.Bundle
+import android.view.View
+import android.widget.EditText
+import android.widget.ImageView
+import androidx.appcompat.app.AppCompatActivity
+import androidx.recyclerview.widget.LinearLayoutManager
+import androidx.recyclerview.widget.RecyclerView
+import ch.epfl.sdp.blindly.R
+import com.google.firebase.auth.FirebaseAuth
+import com.google.firebase.database.*
+import com.google.firebase.database.ktx.getValue
+
+
+class ChatActivity : AppCompatActivity() {
+
+    private lateinit var currentUserId: String
+    private lateinit var matchId: String
+    private lateinit var chatId: String
+    private lateinit var chatReference: DatabaseReference
+
+    private var chatMessages: ArrayList<Message>? = arrayListOf()
+    var mChatLayoutManager = LinearLayoutManager(this)
+
+    /**
+     * Gets the current user's uid and also uid of the matched user via Bundle;
+     * from those it forms a chatID which we'll use to refer in the Realtime Database
+     * Using the chatId we get the messages from Realtime Database via receiveMessages()
+     *
+     * Note: We need to give the url to the getInstance() function as our database is not
+     * located in the USA
+     *
+     * @param savedInstanceState
+     */
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_chat)
+        currentUserId = FirebaseAuth.getInstance().currentUser.uid
+
+        val bundle = this.intent.extras
+        matchId = if (bundle != null) {
+            bundle.getString("matchId").toString()
+        } else {
+            //for manual testing
+            //matchId = "QDs8Hu6kWAb7SSdGQUbDllJXX3y2" //ct
+            "kq37EpNoqqPA3o6vjAFFuRUgDkE2" //cgngr
```

Comment:
Why not just put the test id in the bundle why testing? I don't think having test code in production is a good idea.

Reply:
You are right, I corrected this.